repository/src/main/java/com/epam/esm/certificate/dao/CertificateDao.java

```diff
@@ -0,0 +1,86 @@
+package com.epam.esm.certificate.dao;
+
+import com.epam.esm.certificate.mapper.CertificateMapper;
+import com.epam.esm.certificate.model.Certificate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class CertificateDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    private final String SQL_FIND = "select * from certificate where id = ?";
```

Comment:
It is a bad practice to use '*' instead of specific fields in sql query, even we need all fields